app/src/main/java/com/example/dsublett/monsterfetch/MainActivity.kt

```diff
@@ -3,30 +3,60 @@ package com.example.dsublett.monsterfetch
 import android.os.AsyncTask
 import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
-import android.widget.TextView
+import android.support.v7.widget.LinearLayoutManager
+import android.view.View
+import com.squareup.moshi.Moshi
+import java.net.URL
 import kotlinx.android.synthetic.main.activity_main.*
-import java.net.*
 
-
-class MainActivity : AppCompatActivity() {
+private class MainActivity : AppCompatActivity() {
+    private val monsters: ArrayList<String> = ArrayList() // Data set for recycler view
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        val responseTextView = findViewById<TextView>(R.id.responseTextView)
-        fetchBtn.setOnClickListener {
-            AsyncTaskExample(this).execute()
-        }
+        this.rv_monster_list.layoutManager = LinearLayoutManager(this)
+        this.rv_monster_list.adapter = MonsterAdapter(monsters, this)
+    }
+    fun updateDataSet(view: View) {
+        // Asynchronously retrieve list of monsters, parse into List, add to data set
+        UpdateMonsterList(this.monsters).execute()
+    }
+    fun clearDataSet(view: View) {
+        ClearMonsterList(this.monsters).execute()
+    }
+    fun updateView(view: View) {
+        // Notify view that data set changed
+        this.rv_monster_list.adapter?.notifyDataSetChanged()
     }
-
 }
 
-class AsyncTaskExample(private var activity: MainActivity?) : AsyncTask<String, String, String>() {
-    override fun doInBackground(vararg p0: String?): String {
-        return URL("http://www.dnd5eapi.co/api/monsters").readText()
+private class UpdateMonsterList(val monsterList: ArrayList<String>)
```

Comment:
take a closure (callback) instead of a list.
`callback: ((listOfMonsters: List<Monster) -> Unit)`